### DISCUSSION TITLE
Validate Kustomization without cluster (from CI)

### DISCUSSION THREAD

**Discussion by dmitriishaburov** [fluxcd/flux2#2571] [Q&A] [answered]
Hello!

I was wondering - it there any way to validate / build Kustomization for Flux without any access to Kubernetes?

Usecase: I'd like to check in GitHub actions on PR that Kustomization is valid, to avoid possible build issue after PR is merged (broken paths, invalid kustomizations, `namespace not specified` etc)

I see the `flux build kustomization` that probably does the same as Flux during apply, but it requires access to Kubernetes API.

**Comment by stefanprodan** [comment, marked as answer]
Some things can be done statically but not all, since validation depends on admission controllers, registered CRDs, etc.

For static validation in CI we offer a script and a GH Action here: 
- https://github.com/fluxcd/flux2-kustomize-helm-example/blob/main/scripts/validate.sh
- https://github.com/fluxcd/flux2-kustomize-helm-example/blob/main/.github/workflows/test.yaml
- https://github.com/fluxcd/flux2-kustomize-helm-example/tree/main/.github/actions/tools

**Reply by dmitriishaburov** [reply]
Thanks, tested validate.sh, it does some checks.
Although not catching the `namespace not specified` problem, but that's a good start.
